### ISSUE TITLE
Improve Null pointer checking

### ISSUE THREAD

**Issue by ruudk** [kalessil/phpinspectionsea#593]
The awesome plugin is warning me about a possible null pointer. But that's not really the case since I'm using an Assert on `hasPlatform` before calling `getPlatform`. 

Example:
```php
// Some object has this method: function getPlatform() : ?Platform;

\Webmozart\Assert\Assert::true($object->hasPlatform());
$object->getPlatform()->callSomeFunction();
```
will result in
<img width="374" alt="screen shot 2017-10-18 at 14 51 18" src="https://user-images.githubusercontent.com/104180/31719545-1db377f6-b414-11e7-97d9-c5d49653dbcc.png">

This gives me 2 ideas to improve this:

1) Would it be possible to see if `has*` is called/checked before calling the `get*` method?

2) Since Webmozart\Assert and Beberlei\Assert are really popular in the PHP community, would it be an idea to detect them?

**Comment by kalessil** [comment]
Looks like `getPlatform` declares return type can be a null and the inspection just point to this fact.
Commonly used practice is if `hasSomething`-getSomething` methods pair used, then `getSomething` return `something` or throws an exception.

IWe had a similar report already, but I cannot find if for linking.
